arangod/Cluster/RebootTracker.cpp

```diff
@@ -66,296 +84,151 @@ RebootTracker::RebootTracker(RebootTracker::SchedulerPointer scheduler)
 #endif
 }
 
-void RebootTracker::updateServerState(
-    containers::FlatHashMap<ServerID, RebootId> const& state) {
-  MUTEX_LOCKER(guard, _mutex);
-
+void RebootTracker::updateServerState(State state) {
+  std::lock_guard guard{_mutex};
   LOG_TOPIC("77a6e", DEBUG, Logger::CLUSTER)
-      << "updating reboot server state from " << _rebootIds << " to " << state;
-
-  // Call cb for each iterator.
-  auto for_each_iter = [](auto begin, auto end, auto cb) {
-    auto it = begin;
-    decltype(it) next;
-    while (it != end) {
-      // save next iterator now, in case cb invalidates it.
-      next = std::next(it);
-      cb(it);
-      it = next;
-    }
-  };
-
-  // For all known servers, look whether they are changed or were removed
-  for_each_iter(_rebootIds.begin(), _rebootIds.end(), [&](auto const curIt) {
-    auto const& serverId = curIt->first;
-    auto& oldRebootId = curIt->second;
-    auto const& newIt = state.find(serverId);
-
+      << "updating reboot server state from " << _state << " to " << state;
+  for (auto it = _callbacks.begin(); it != _callbacks.end();) {
+    auto next = std::next(it);
+    auto newIt = state.find(it->first);
+    TRI_ASSERT(!it->second.empty());  // we should remove it, when it was empty
     if (newIt == state.end()) {
-      // Try to schedule all callbacks for serverId.
-      // If that didn't throw, erase the entry.
-      LOG_TOPIC("88858", INFO, Logger::CLUSTER)
-          << "Server " << serverId << " removed, aborting its old jobs now.";
-      scheduleAllCallbacksFor(serverId);
-      auto it = _callbacks.find(serverId);
-      if (it != _callbacks.end()) {
-        TRI_ASSERT(it->second.empty());
-        _callbacks.erase(it);
-      }
-      _rebootIds.erase(curIt);
+      takeCallbacksFor(it);
     } else {
-      TRI_ASSERT(serverId == newIt->first);
-      auto const& newRebootId = newIt->second;
-      TRI_ASSERT(oldRebootId <= newRebootId);
-      if (oldRebootId < newRebootId) {
-        LOG_TOPIC("88857", INFO, Logger::CLUSTER)
-            << "Server " << serverId
-            << " gone or rebooted, aborting its old jobs now.";
-        // Try to schedule all callbacks for serverId older than newRebootId.
-        // If that didn't throw, erase the entry.
-        scheduleCallbacksFor(serverId, newRebootId);
-        oldRebootId = newRebootId;
-      }
+      auto const toRebootId = newIt->second;
+      takeCallbacksFor(it, toRebootId);
     }
```

Comment:
Them are useless

Reply:
Could you help me understand why it's not worth having these log messages? Is it that they convey too little information, or are never actually printed?

Reply:
@apetenchea
I have a two reason
1) Now I iterate for callbacks, not for old state, I never iterate for old or new state in code, so I can't re-add this log messages
2) Also I think these information are already printed in this line https://github.com/arangodb/arangodb/blob/1fc33515b454105b4b59258940af661536105065/arangod/Cluster/RebootTracker.cpp#L90

Reply:
> Them are useless

I beg to differ. They've been regularly helpful in the past for debugging, also on customer's sites.

> Now I iterate for callbacks, not for old state, I never iterate for old or new state in code, so I can't re-add this log messages

`s/I can't re-add this log messages/it's slightly more inconvenient to re-add them/`

> Also I think these information are already printed in this line (was in old code) 
That message is on DEBUG level, and for good reason, because it is printed way more often.